lib/algo/DMDc.h

```diff
@@ -291,12 +299,13 @@ class DMDc
     /**
      * @brief Add the state offset when predicting the solution.
      */
-    virtual void addOffset(Vector*& result);
+    virtual void addOffset(Vector & result);
 
     /**
      * @brief Get the snapshot matrix contained within d_snapshots.
      */
-    const Matrix* createSnapshotMatrix(std::vector<Vector*> snapshots);
+    std::unique_ptr<const Matrix> createSnapshotMatrix(std::vector<Vector*>
```

Comment:
Should `snapshots` be passed by reference?

Reply:
Done.